src/emit.iced

```diff
@@ -60,22 +69,39 @@ exports.GoEmitter = class GoEmitter
   tab : () -> @_tabs++
   untab : () -> @_tabs--
 
+  make_map_type : ({t}) ->
+    key = if t.keys? then @emit_field_type(t.keys).type else "string"
+    "map[#{key}]" + @emit_field_type(t.values).type
+
   emit_field_type : (t, {pointed} = {}) ->
     optional = !!pointed
     type = if typeof(t) is 'string' then @go_primitive_type(t)
     else if typeof(t) is 'object'
-      if Array.isArray(t) and not t[0]?
-        optional = true
-        "*" + @go_primitive_type(t[1])
-      else if t.type is "array" then "[]" + @go_primitive_type(t.items)
+      if Array.isArray(t)
+        if not t[0]?
+          optional = true
+          "*" + @emit_field_type(t[1]).type
+        else
+          "ERROR"
+      else if t.type is "array"
+        "[]" + @emit_field_type(t.items).type
       else if t.type is "map"
-        key = if t.keys? then @go_lint_capitalize(@emit_field_type(t.keys).type) else "string"
-        "map[#{key}]" + @go_lint_capitalize(@emit_field_type(t.values).type)
+        @make_map_type { t }
       else "ERROR"
     else "ERROR"
     type = "*" + type if pointed
     { type , optional }
 
+  emit_typedef_deep_copy : ({t}) ->
+    type = t.name
+    receiver = "o"
+    @output "func (#{receiver} #{type}) DeepCopy() #{type} {"
```

Comment:
Any particular reason DeepCopy takes a value receiver? Might be slightly more useful and (maybe I'm making this up) faster to have a pointer receiver that also handles `nil.DeepCopy() -> nil`.